superset/utils/cache_manager.py

```diff
@@ -14,73 +14,78 @@
-            app,
-            {
-                "CACHE_DEFAULT_TIMEOUT": app.config["CACHE_DEFAULT_TIMEOUT"],
-                **app.config["CACHE_CONFIG"],
-            },
-        )
-        self._data_cache.init_app(
-            app,
-            {
-                "CACHE_DEFAULT_TIMEOUT": app.config["CACHE_DEFAULT_TIMEOUT"],
-                **app.config["DATA_CACHE_CONFIG"],
-            },
-        )
-        self._thumbnail_cache.init_app(
-            app,
-            {
-                "CACHE_DEFAULT_TIMEOUT": app.config["CACHE_DEFAULT_TIMEOUT"],
-                **app.config["THUMBNAIL_CACHE_CONFIG"],
-            },
-        )
-        self._filter_state_cache.init_app(
-            app,
-            {
-                "CACHE_DEFAULT_TIMEOUT": app.config["CACHE_DEFAULT_TIMEOUT"],
-                **app.config["FILTER_STATE_CACHE_CONFIG"],
-            },
+        if app.debug:
+            self._default_cache_config = {
+                "CACHE_TYPE": "SimpleCache",
+                "CACHE_THRESHOLD": math.inf,
+            }
+        else:
+            self._default_cache_config = {}
+
+        default_timeout = app.config.get("CACHE_DEFAULT_TIMEOUT")
+        if default_timeout is not None:
+            self._default_cache_config["CACHE_DEFAULT_TIMEOUT"] = default_timeout
+            logger.warning(
+                _(
+                    "The global config flag `CACHE_DEFAULT_TIMEOUT` has been "
+                    "deprecated and will be removed in Superset 2.0. Please set "
+                    "default cache options in the `DEFAULT_CACHE_CONFIG` parameter"
+                ),
+            )
+        self._default_cache_config = {
+            **self._default_cache_config,
+            **app.config["DEFAULT_CACHE_CONFIG"],
+        }
+
+        self._init_cache(app, self.cache, "CACHE_CONFIG")
+        self._init_cache(app, self.data_cache, "DATA_CACHE_CONFIG")
+        self._init_cache(app, self.thumbnail_cache, "THUMBNAIL_CACHE_CONFIG")
+        self._init_cache(
+            app, self.filter_state_cache, "FILTER_STATE_CACHE_CONFIG", required=True
         )
-        self._explore_form_data_cache.init_app(
+        self._init_cache(
             app,
-            {
-                "CACHE_DEFAULT_TIMEOUT": app.config["CACHE_DEFAULT_TIMEOUT"],
-                **app.config["EXPLORE_FORM_DATA_CACHE_CONFIG"],
-            },
+            self.explore_form_data_cache,
+            "EXPLORE_FORM_DATA_CACHE_CONFIG",
+            required=True,
         )
-
-    @property
-    def data_cache(self) -> Cache:
-        return self._data_cache
-
-    @property
-    def cache(self) -> Cache:
-        return self._cache
-
-    @property
-    def thumbnail_cache(self) -> Cache:
-        return self._thumbnail_cache
-
-    @property
-    def filter_state_cache(self) -> Cache:
-        return self._filter_state_cache
-
-    @property
-    def explore_form_data_cache(self) -> Cache:
-        return self._explore_form_data_cache
```

Comment:
This seemed very Java getter/setter-like with little utility (I assume a remnant from old times)